src/Simulations/simulation.jl

```diff
@@ -88,7 +93,8 @@ function Simulation(model; Δt,
                      0.0,
                      false,
                      false,
-                     verbose)
+                     verbose,
+                     Float64(minimum_relative_step))
```

Comment:
why Float64?

Reply:
I was following the other floats in the `Simulation` constructor, which are also converted to `Float64`. I can't remember the PR where this was decided, but it minimizes errors in time-step alignment. The error that this PR solves is an example of this type of error .

Reply:
Probably we should make it optional. 
This will not work if we want to support Metal architectures that do not want Float64.

Reply:
That sounds good. Although I think we should do that on another PR, since there are other things that follow this convention. For this PR I think we should follow what's already there.

Reply:
Well now that I see it better, this never goes in a kernel, so it should be fine

Reply:
should be easy to change these if we need to though as stated I don't know if we do for Metal since they aren't kernel variables